### PR TITLE
Migration wizard Refresh 11th Feb 2021 

### PR DESCRIPTION
**Pull request by aasimkhan30**
Accounts Selection Page: 
image

Temporary Targets Page:
image

Database Backup Page:
image

Migration Controller Page
image

Create migration controller dialog
Initial State
image

After Clicking Create and IR is not registered
image

When IR is registered
image

When done is clicked
image

Summary Page
image

### CODE REVIEW

File: extensions/sql-migration/src/api/azure.ts [hunk 1]
```diff
@@ -175,3 +226,62 @@ export interface GetMigrationControllerAuthKeysResult {
...
+	cpuUtilization: number,
+	nodeName: string
+	receivedBytes: number
+	sentBytes: number
+}
+
+export interface StartDatabaseMigrationRequest {
+	location: string,
+	properties: {
+		SourceDatabaseName: string,
```
Comment (kburtram) [inline extensions/sql-migration/src/api/azure.ts:253]: do we need to use this casing style to work with the API?  or can we use the normal JS camelCase style.

File: extensions/sql-migration/src/models/strings.ts [hunk 2]
```diff
@@ -38,9 +38,10 @@ export const CONGRATULATIONS = localize('sql.migration.generic.congratulations',
 
 
 // Accounts page
-export const ACCOUNTS_SELECTION_PAGE_TITLE = localize('sql.migration.wizard.account.title', "Select your Azure account");
+export const ACCOUNTS_SELECTION_PAGE_TITLE = localize('sql.migration.wizard.account.title', "Azure Account");
+export const ACCOUNTS_SELECTION_PAGE_DESCRIPTION = localize('sql.migration.wizard.account.description', "Select an Azure account linked to Azure Data Studio or link one now.");
 export const ACCOUNT_SELECTION_PAGE_NO_LINKED_ACCOUNTS_ERROR = localize('sql.migration.wizard.account.noaccount.error', "There is no linked account. Please add an account.");
-export const ACCOUNT_ADD_BUTTON_LABEL = localize('sql.migration.wizard.account.add.button.label', "Add account");
+export const ACCOUNT_LINK_BUTTON_LABEL = localize('sql.migration.wizard.account.add.button.label', "Link account");
```
Comment (Charles-Gagnon) [inline extensions/sql-migration/src/models/strings.ts:44]: Where did this text come from? I don't think we use `Link account` anywhere else. Typically we have a `Sign in` button instead.

That being said I'm not against the idea - but we should be consistent across all the wizards.

File: extensions/sql-migration/src/wizard/accountsSelectionPage.ts [hunk 3]
```diff
@@ -30,32 +31,36 @@ export class AccountsSelectionPage extends MigrationWizardPage {
...
-			if ((<azdata.CategoryValue>c.value).displayName === constants.ACCOUNT_SELECTION_PAGE_NO_LINKED_ACCOUNTS_ERROR) {
-				this.wizard.message = {
-					text: constants.ACCOUNT_SELECTION_PAGE_NO_LINKED_ACCOUNTS_ERROR,
-					level: azdata.window.MessageLevel.Error
-				};
-				return false;
-			}
-			return true;
-		}).component();
+		this._azureAccountsDropdown = view.modelBuilder.dropDown()
```
Comment (Charles-Gagnon) [inline extensions/sql-migration/src/wizard/accountsSelectionPage.ts:34]: @alanrenmsft We should really consider adding Azure account/sub/RG dropdowns to the extension API. It's a pretty common control and having to re-implement it each time is not ideal.

File: extensions/sql-migration/src/wizard/createMigrationControllerDialog.ts [hunk 4]
```diff
@@ -278,133 +265,115 @@ export class CreateMigrationControllerDialog {
...
 
-		const manualSetupSecondDescription = this._view.modelBuilder.text().withProps({
-			value: constants.CONTROLLER_OPTION2_STEP2
+		const irSetupStep2Text = this._view.modelBuilder.text().withProps({
+			value: constants.CONTROLLER_STEP2
 		}).component();
 
-		const connectionStatusTitle = this._view.modelBuilder.text().withProps({
-			value: constants.CONTROLLER_CONNECTION_STATUS,
+		const irSetupStep3Text = this._view.modelBuilder.hyperlink().withProps({
```
Comment (Charles-Gagnon) [inline extensions/sql-migration/src/wizard/createMigrationControllerDialog.ts:293]: I'd suggest adding a little spacing between the table and this link - they're a little too close I think

**Review by Charles-Gagnon — APPROVED** [review]

File: extensions/sql-migration/src/api/azure.ts (hunk 1 above)
Comment (aasimkhan30) [reply extensions/sql-migration/src/api/azure.ts:253]: Yes, this is the style followed by API.

Comment (kburtram) [reply extensions/sql-migration/src/api/azure.ts:253]: many of the API frameworks will do auto conversion between one naming style and the other.  For example, when communicating with the C# backend the we can use JS style names in this codebase and C# style names in that codebase, and the conversion is transparent.  Have you confirmed this isn't the case with this API?

**Review by kburtram — APPROVED** [review]
LGTM.  Charles left several comments to consider before merging.

File: extensions/sql-migration/src/models/strings.ts (hunk 2 above)
Comment (aasimkhan30) [reply extensions/sql-migration/src/models/strings.ts:44]: This was a change the UX team recently made in the mockups.

**Pull request merged**